src/main/java/com/google/sps/api/Utils.java

```diff
@@ -47,20 +47,21 @@
 
   /** Global instance of the JSON factory. */
   public static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+  public static final String CLIENT_SECRETS_PATH = "/client_secrets.json";
+  public static final String OFFLINE_ACCESS_TYPE = "offline";
 
   private static GoogleClientSecrets clientSecrets = null;
 
   /**
-   * Load the application's client secrets from the resources/client_secrets.json file
+   * Loads the application's client secrets from the resources/client_secrets.json file
    */
   public static GoogleClientSecrets getClientCredential() throws IOException {
     if (clientSecrets == null) {
       clientSecrets = GoogleClientSecrets.load(JSON_FACTORY,
-          new InputStreamReader(Utils.class.getResourceAsStream("/client_secrets.json")));
+          new InputStreamReader(Utils.class.getResourceAsStream(CLIENT_SECRETS_PATH)));
       Preconditions.checkArgument(!clientSecrets.getDetails().getClientId().startsWith("Enter ")
           && !clientSecrets.getDetails().getClientSecret().startsWith("Enter "),
-          "Download client_secrets.json file from https://code.google.com/apis/console/"
-          + "?api=calendar into calendar-appengine-sample/src/main/resources/client_secrets.json");
```

Comment:
This code works if there is a file checked into the repository that contains:
```
{
  "web": {
    "client_id": "Enter Client ID",
    "client_secret": "Enter Client Secret"
  }
}
```
Otherwise, there is a NullPointer exception.
So please either include sample file in repository (with .gitignore entry as well) or change the condition to check for NullPointerException when trying to read non-existing file.

Reply:
(I see I've commented on the older version of the code, but it still works in the same way)

Reply:
Yes yes, I was thinking about this. Which approach would be better in your opinion?
I would go with the sample file.

Reply:
I think lack of file and NullPointerException handling is better because you don't check into the repo a file that needs to be overridden by every user (and therefore needs to be gitignored, and gitignored file that is in the repo is somehow strange).
Plus adding a prerequisite step to the README that explains how to get credentials file.
And .gitignore entry.

Reply:
.gitignor is I think already added

Reply:
You're right!